docs/components/table.md

```diff
@@ -712,7 +858,53 @@ export default {
         'data-testid': 'row-item'
       }
     },
-    defaultSorter
+    defaultSorter,
+    cellAttrsFn ({ headerKey, row, rowIndex, colIndex }) {
+      /**
+       * Sets cell background color based on data returned in
+       * the row parameter and the index of the cell
+       */
+      const backgroundColor = () => {
+        if (row.company && row.company === 'Google') {
+          if (colIndex === 0) {
+            return '#4285F4'
+          } else if (colIndex === 1) {
+            return '#DB4437'
+          } else {
+            return '#F4B400'
+          }
+        }
+
+        return ''
+      }
+
+      /**
+       * Sets cell max-width based on the column headerKey
+       */
+      const colWidth = () => {
+        if (headerKey === 'description') {
+          return '50ch'
+        } else if (headerKey === 'name') {
+          return '22ch'
+        }
+
+        return '25ch'
+      }
+
+      /**
+       * Returns an object of attributes to be applied to cells
+       */
+      return {
+        class: {
+          'truncate': headerKey === 'description' || headerKey === 'name',
+        },
+        'datatest-id': `row-${rowIndex + 1}-col-${headerKey}`,
+        style: {
+          'maxWidth': colWidth(),
+          'backgroundColor': backgroundColor(),
```

Comment:
I think this could be difficult to read as an example as it creates unnecessary intermediary functions, and else condition with early return.
```
style:{
  'maxWidth': headerKey==='description' ? '50ch' : headerKey === 'name' ? '22ch' : '25ch'
}
```
This approach has less potential for side effect from a change in conditions or use of variables out of scope.

Aside from that, KTable traditionally used slots and classes, which I feel discourages css in js. I feel like this might not be the most idiomatic vue but not a blocking concern. Just my two cents.

Reply:
> Would like to see some prior art from another component library. In the past segmentio's evergreen has been an inspiration for component API decisions. https://evergreen.segment.com/components/table
> In this case I didn't find anything, but I'm curious if this change is based on similar inspiration?

@jackkav thanks for your review and feedback. I've removed the unnecessary intermediary function from `maxWidth` in the example. I made this change based on the `rowAttributes` KTable prop. In our use case, we believe adding the ability to customize the `max-width` value for columns will allow us to utilize more of the currently empty space in our data tables. I felt that passing information about the row and column would enable developers to customize with more specificity. As an optional prop, it's there for developers to leverage, but can simply be ignored if they don't need it.

As for examples from another component library, these are the closest examples I could find from Segment:
1. https://github.com/segmentio/evergreen/blob/master/src/table/src/TextTableCell.js#L29
2. https://github.com/segmentio/evergreen/blob/master/src/typography/src/Text.js#L32-L36
3. https://github.com/segmentio/evergreen/blob/master/src/table/src/TableCell.js#L106

Others (these are just similar examples, not identical use cases):
1. Microsoft Fluent UI: https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/customitemcolumns.
2. Elastic EUI: https://elastic.github.io/eui/#/tabular-content/data-grid-styling-and-control